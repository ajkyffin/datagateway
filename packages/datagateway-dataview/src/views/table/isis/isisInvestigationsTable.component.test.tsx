import * as React from 'react';
import ISISInvestigationsTable from './isisInvestigationsTable.component';
import { initialState as dgDataViewInitialState } from '../../../state/reducers/dgdataview.reducer';
import configureStore from 'redux-mock-store';
import type { StateType } from '../../../state/app.types';
import {
  dGCommonInitialState,
  type Investigation,
  useAddToCart,
  useCart,
  useInvestigationDetails,
  useInvestigationSizes,
  useISISInvestigationCount,
  useISISInvestigationIds,
  useISISInvestigationsInfinite,
  useRemoveFromCart,
} from 'datagateway-common';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import { createMemoryHistory, type History } from 'history';
import {
  applyDatePickerWorkaround,
  cleanupDatePickerWorkaround,
  findAllRows,
  findCellInRow,
  findColumnIndexByName,
  findRowAt,
} from '../../../setupTests';
import {
  render,
  type RenderResult,
  screen,
  waitFor,
  within,
} from '@testing-library/react';
import type { UserEvent } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

jest.mock('datagateway-common', () => {
  const originalModule = jest.requireActual('datagateway-common');

  return {
    __esModule: true,
    ...originalModule,
    useISISInvestigationCount: jest.fn(),
    useISISInvestigationsInfinite: jest.fn(),
    useInvestigationSizes: jest.fn(),
    useISISInvestigationIds: jest.fn(),
    useCart: jest.fn(),
    useAddToCart: jest.fn(),
    useRemoveFromCart: jest.fn(),
    useInvestigationDetails: jest.fn(),
  };
});

describe('ISIS Investigations table component', () => {
  let mockStore;
  let state: StateType;
  let rowData: Investigation[];
  let history: History;
  let replaceSpy: jest.SpyInstance;
  let user: UserEvent;

  const renderComponent = (studyHierarchy = false): RenderResult => {
    const store = mockStore(state);
    return render(
      <Provider store={store}>
        <Router history={history}>
          <QueryClientProvider client={new QueryClient()}>
            <ISISInvestigationsTable
              studyHierarchy={studyHierarchy}
              instrumentId="4"
              instrumentChildId="5"
            />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
  };

  beforeEach(() => {
    rowData = [
      {
        id: 1,
        title: 'Test 1',
        name: 'Test 1',
        summary: 'foo bar',
        visitId: '1',
        doi: 'doi 1',
        size: 1,
        investigationUsers: [
          {
            id: 2,
            role: 'experimenter',
            user: { id: 2, name: 'test', fullName: 'Test experimenter' },
          },
          {
            id: 3,
            role: 'principal_experimenter',
            user: { id: 3, name: 'testpi', fullName: 'Test PI' },
          },
        ],
        studyInvestigations: [
          {
            id: 6,
            study: {
              id: 7,
              pid: 'study pid',
            },
          },
        ],
        startDate: '2019-06-10',
        endDate: '2019-06-11',
      },
    ];
    history = createMemoryHistory();
    replaceSpy = jest.spyOn(history, 'replace');
    user = userEvent.setup();

    mockStore = configureStore([thunk]);
    state = JSON.parse(
      JSON.stringify({
        dgdataview: dgDataViewInitialState,
        dgcommon: dGCommonInitialState,
      })
    );

    (useCart as jest.Mock).mockReturnValue({
      data: [],
      isLoading: false,
    });
    (useISISInvestigationCount as jest.Mock).mockReturnValue({
      data: 0,
    });
    (useISISInvestigationsInfinite as jest.Mock).mockReturnValue({
      data: { pages: [rowData] },
      fetchNextPage: jest.fn(),
    });
    (useInvestigationSizes as jest.Mock).mockReturnValue([
      {
        data: 1,
      },
    ]);
    (useISISInvestigationIds as jest.Mock).mockReturnValue({
      data: [1],
      isLoading: false,
    });
    (useAddToCart as jest.Mock).mockReturnValue({
      mutate: jest.fn(),
      isLoading: false,
    });
    (useRemoveFromCart as jest.Mock).mockReturnValue({
      mutate: jest.fn(),
      isLoading: false,
    });
    (useInvestigationDetails as jest.Mock).mockReturnValue({
      data: [],
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('displays DOI and renders the expected Link ', async () => {
    renderComponent();
    expect(
      await screen.findByRole('link', { name: 'study pid' })
    ).toHaveAttribute('href', 'https://doi.org/study pid');
  });

  it('updates filter query params on text filter', async () => {
    renderComponent();

    const filterInput = await screen.findByRole('textbox', {
      name: 'Filter by investigations.name',
      hidden: true,
    });

    await user.type(filterInput, 'test');

    // user.type inputs the given string character by character to simulate user typing
    // each keystroke of user.type creates a new entry in the history stack
    // so the initial entry + 4 characters in "test" = 5 entries
    expect(history.length).toBe(5);
    expect(history.location.search).toBe(
      `?filters=${encodeURIComponent(
        '{"name":{"value":"test","type":"include"}}'
      )}`
    );

    await user.clear(filterInput);

    expect(history.length).toBe(6);
    expect(history.location.search).toBe('?');
  });

  it('updates filter query params on date filter', async () => {
    applyDatePickerWorkaround();

    renderComponent();

    const filterInput = await screen.findByRole('textbox', {
      name: 'investigations.start_date filter from',
    });

    await user.type(filterInput, '2019-08-06');

    expect(history.length).toBe(2);
    expect(history.location.search).toBe(
      `?filters=${encodeURIComponent(
        '{"startDate":{"startDate":"2019-08-06"}}'
      )}`
    );

    await user.clear(filterInput);

    expect(history.length).toBe(3);
    expect(history.location.search).toBe('?');

    cleanupDatePickerWorkaround();
  });

  it('uses default sort', () => {
    renderComponent();
    expect(history.length).toBe(1);
    expect(replaceSpy).toHaveBeenCalledWith({
      search: `?sort=${encodeURIComponent('{"startDate":"desc"}')}`,
    });
  });

  it('updates sort query params on sort', async () => {
    renderComponent();

    await user.click(
      await screen.findByRole('button', { name: 'investigations.title' })
    );

    expect(history.length).toBe(2);
    expect(history.location.search).toBe(
      `?sort=${encodeURIComponent('{"title":"asc"}')}`
    );
  });

  it('calls addToCart mutate function on unchecked checkbox click', async () => {
    const addToCart = jest.fn();
    (useAddToCart as jest.Mock).mockReturnValue({
      mutate: addToCart,
      loading: false,
    });
    renderComponent();

    await user.click(
      await screen.findByRole('checkbox', { name: 'select row 0' })
    );

    expect(addToCart).toHaveBeenCalledWith([1]);
  });

  it('calls removeFromCart mutate function on checked checkbox click', async () => {
    (useCart as jest.Mock).mockReturnValue({
      data: [
        {
          entityId: 1,
          entityType: 'investigation',
          id: 1,
          name: 'test',
          parentEntities: [],
        },
      ],
      isLoading: false,
    });

    const removeFromCart = jest.fn();
    (useRemoveFromCart as jest.Mock).mockReturnValue({
      mutate: removeFromCart,
      loading: false,
    });

    renderComponent();

    await user.click(
      await screen.findByRole('checkbox', { name: 'select row 0' })
    );

    expect(removeFromCart).toHaveBeenCalledWith([1]);
  });

  it('selected rows only considers relevant cart items', async () => {
    (useCart as jest.Mock).mockReturnValueOnce({
      data: [
        {
          entityId: 2,
          entityType: 'investigation',
          id: 1,
          name: 'test',
          parentEntities: [],
        },
        {
          entityId: 1,
          entityType: 'dataset',
          id: 2,
          name: 'test',
          parentEntities: [],
        },
      ],
      isLoading: false,
    });

    renderComponent();

    const selectAllCheckbox = await screen.findByRole('checkbox', {
      name: 'select all rows',
    });

    expect(selectAllCheckbox).not.toBeChecked();
    expect(selectAllCheckbox).toHaveAttribute('data-indeterminate', 'false');
  });

  it('no select all checkbox appears and no fetchAllIds sent if selectAllSetting is false', async () => {
    state.dgdataview.selectAllSetting = false;
    renderComponent();
    await waitFor(() => {
      expect(
        screen.queryByRole('button', { name: 'select all rows' })
      ).toBeNull();
    });
  });

  it('displays details panel when expanded', async () => {
    renderComponent();
    await user.click(
      await screen.findByRole('button', { name: 'Show details' })
    );
    expect(
      await screen.findByTestId('investigation-details-panel')
    ).toBeTruthy();
  });

  it('renders details panel with datasets link and can navigate', async () => {
    renderComponent();

    await user.click(
      await screen.findByRole('button', { name: 'Show details' })
    );

    await user.click(
      await screen.findByRole('tab', {
        name: 'investigations.details.datasets',
      })
    );

    expect(history.location.pathname).toBe(
      '/browse/instrument/4/facilityCycle/5/investigation/1/dataset'
    );
  });

  it('renders title and DOI as links', async () => {
    renderComponent();
    expect(await screen.findByRole('link', { name: 'Test 1' })).toHaveAttribute(
      'href',
      '/browse/instrument/4/facilityCycle/5/investigation/1'
    );
    expect(
      await screen.findByRole('link', { name: 'study pid' })
    ).toHaveAttribute('href', 'https://doi.org/study pid');
  });

  it('renders title and DOI as links in StudyHierarchy', async () => {
    renderComponent(true);
    expect(await screen.findByRole('link', { name: 'Test 1' })).toHaveAttribute(
      'href',
      '/browseStudyHierarchy/instrument/4/study/5/investigation/1'
    );
    expect(
      await screen.findByRole('link', { name: 'study pid' })
    ).toHaveAttribute('href', 'https://doi.org/study pid');
  });

  it('displays the correct user as the PI ', async () => {
    renderComponent();

    const piColumnIndex = await findColumnIndexByName(
      'investigations.principal_investigators'
    );

    const row = await findRowAt(0);

    expect(
      await findCellInRow(row, { columnIndex: piColumnIndex })
    ).toHaveTextContent('Test PI');
  });

  it('gracefully handles empty Study Investigation and investigationUsers', async () => {
    (useISISInvestigationsInfinite as jest.Mock).mockReturnValue({
      data: {
        pages: [
          {
            ...rowData[0],
            investigationUsers: [],
            studyInvestigations: [],
          },
        ],
      },
      fetchNextPage: jest.fn(),
    });

    renderComponent();

    const rows = await screen.findAllByRole('row');
    // 2 rows expected, 1 for the header row, and 1 for the items in rowData.
    expect(rows).toHaveLength(2);
  });

  it('gracefully handles missing Study from Study Investigation object and missing User from investigationUsers object', async () => {
    (useISISInvestigationsInfinite as jest.Mock).mockClear();
    (useISISInvestigationsInfinite as jest.Mock).mockReturnValue({
      data: {
        pages: [
          {
            ...rowData[0],
            investigationUsers: [
              {
                id: 1,
              },
            ],
            studyInvestigations: [
              {
                id: 6,
              },
            ],
          },
        ],
      },
      fetchNextPage: jest.fn(),
    });

    renderComponent();

    const doiColumnIndex = await findColumnIndexByName('investigations.doi');
    const piColumnIndex = await findColumnIndexByName(
      'investigations.principal_investigators'
    );

    // verify that the doi cell and the principal investigator cell in the row are empty
    const row = await findRowAt(0);

    expect(
      await findCellInRow(row, { columnIndex: doiColumnIndex })
    ).toHaveTextContent('');
    expect(
      await findCellInRow(row, { columnIndex: piColumnIndex })
    ).toHaveTextContent('');
  });

  it('renders actions correctly', async () => {
    renderComponent();

    // find the action column
    const actionsColumnIndex = await findColumnIndexByName('Actions');

    // make sure all rows have the download button
    const rows = await findAllRows();
    for (const row of rows) {
      const actionCell = await findCellInRow(row, {
        columnIndex: actionsColumnIndex,
      });
      expect(
        await within(actionCell).findByRole('button', {
          name: 'buttons.download',
        })
      ).toBeInTheDocument();
    }
  });
});
