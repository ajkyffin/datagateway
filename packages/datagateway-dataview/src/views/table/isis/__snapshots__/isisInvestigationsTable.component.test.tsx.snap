// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ISIS Investigations table component renders correctly 1`] = `
Object {
  "actions": Array [
    [Function],
  ],
  "allIds": Array [
    1,
  ],
  "columns": Array [
    Object {
      "cellContentRenderer": [Function],
      "dataKey": "title",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "investigations.title",
    },
    Object {
      "dataKey": "name",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "investigations.name",
    },
    Object {
      "cellContentRenderer": [Function],
      "dataKey": "studyInvestigations.study.pid",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "investigations.doi",
    },
    Object {
      "cellContentRenderer": [Function],
      "dataKey": "size",
      "disableSort": true,
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "investigations.size",
    },
    Object {
      "cellContentRenderer": [Function],
      "dataKey": "investigationUsers.user.fullName",
      "disableSort": true,
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "investigations.principal_investigators",
    },
    Object {
      "dataKey": "startDate",
      "defaultSort": "desc",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "investigations.start_date",
    },
    Object {
      "dataKey": "endDate",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "investigations.end_date",
    },
  ],
  "data": Array [
    Object {
      "doi": "doi 1",
      "endDate": "2019-06-11",
      "id": 1,
      "investigationUsers": Array [
        Object {
          "id": 2,
          "role": "experimenter",
          "user": Object {
            "fullName": "Test experimenter",
            "id": 2,
            "name": "test",
          },
        },
        Object {
          "id": 3,
          "role": "principal_experimenter",
          "user": Object {
            "fullName": "Test PI",
            "id": 3,
            "name": "testpi",
          },
        },
      ],
      "name": "Test 1",
      "size": 1,
      "startDate": "2019-06-10",
      "studyInvestigations": Array [
        Object {
          "id": 6,
          "study": Object {
            "id": 7,
            "pid": "study pid",
          },
        },
      ],
      "summary": "foo bar",
      "title": "Test 1",
      "visitId": "1",
    },
  ],
  "detailsPanel": [Function],
  "disableSelectAll": false,
  "loadMoreRows": [Function],
  "loading": false,
  "onCheck": [MockFunction],
  "onSort": [Function],
  "onUncheck": [MockFunction],
  "selectedRows": Array [],
  "sort": Object {
    "startDate": "desc",
  },
  "totalRowCount": 0,
}
`;

exports[`ISIS Investigations table component renders title, name and DOI as links 1`] = `
<ForwardRef(Link)
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  data-testid="isis-investigations-table-title"
  to="/browse/instrument/4/facilityCycle/5/investigation/1"
>
  <MuiLinkRoot
    className="MuiLink-root MuiLink-underlineAlways"
    color="primary"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    data-testid="isis-investigations-table-title"
    onBlur={[Function]}
    onFocus={[Function]}
    ownerState={
      Object {
        "children": "Test 1",
        "color": "primary",
        "component": Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        },
        "data-testid": "isis-investigations-table-title",
        "focusVisible": false,
        "to": "/browse/instrument/4/facilityCycle/5/investigation/1",
        "underline": "always",
        "variant": "inherit",
      }
    }
    to="/browse/instrument/4/facilityCycle/5/investigation/1"
    variant="inherit"
  >
    <Noop />
    <ForwardRef(Typography)
      className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-testid="isis-investigations-table-title"
      onBlur={[Function]}
      onFocus={[Function]}
      to="/browse/instrument/4/facilityCycle/5/investigation/1"
      variant="inherit"
    >
      <MuiTypographyRoot
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        data-testid="isis-investigations-table-title"
        onBlur={[Function]}
        onFocus={[Function]}
        ownerState={
          Object {
            "align": "inherit",
            "children": "Test 1",
            "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
            "color": "primary.main",
            "component": Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
            "data-testid": "isis-investigations-table-title",
            "gutterBottom": false,
            "noWrap": false,
            "onBlur": [Function],
            "onFocus": [Function],
            "paragraph": false,
            "sx": Object {
              "color": "primary.main",
            },
            "to": "/browse/instrument/4/facilityCycle/5/investigation/1",
            "variant": "inherit",
            "variantMapping": Object {
              "body1": "p",
              "body2": "p",
              "h1": "h1",
              "h2": "h2",
              "h3": "h3",
              "h4": "h4",
              "h5": "h5",
              "h6": "h6",
              "inherit": "p",
              "subtitle1": "h6",
              "subtitle2": "h6",
            },
          }
        }
        sx={
          Object {
            "color": "primary.main",
          }
        }
        to="/browse/instrument/4/facilityCycle/5/investigation/1"
      >
        <Noop />
        <Link
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
          data-testid="isis-investigations-table-title"
          onBlur={[Function]}
          onFocus={[Function]}
          to="/browse/instrument/4/facilityCycle/5/investigation/1"
        >
          <LinkAnchor
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            data-testid="isis-investigations-table-title"
            href="/browse/instrument/4/facilityCycle/5/investigation/1"
            navigate={[Function]}
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              data-testid="isis-investigations-table-title"
              href="/browse/instrument/4/facilityCycle/5/investigation/1"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Test 1
            </a>
          </LinkAnchor>
        </Link>
      </MuiTypographyRoot>
    </ForwardRef(Typography)>
  </MuiLinkRoot>
</ForwardRef(Link)>
`;

exports[`ISIS Investigations table component renders title, name and DOI as links 2`] = `"Test 1"`;

exports[`ISIS Investigations table component renders title, name and DOI as links 3`] = `
<ForwardRef(Link)
  data-testid="isis-investigation-table-doi-link"
  href="https://doi.org/study pid"
>
  <MuiLinkRoot
    className="MuiLink-root MuiLink-underlineAlways"
    color="primary"
    component="a"
    data-testid="isis-investigation-table-doi-link"
    href="https://doi.org/study pid"
    onBlur={[Function]}
    onFocus={[Function]}
    ownerState={
      Object {
        "children": "study pid",
        "color": "primary",
        "component": "a",
        "data-testid": "isis-investigation-table-doi-link",
        "focusVisible": false,
        "href": "https://doi.org/study pid",
        "underline": "always",
        "variant": "inherit",
      }
    }
    variant="inherit"
  >
    <Noop />
    <ForwardRef(Typography)
      className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
      color="primary"
      component="a"
      data-testid="isis-investigation-table-doi-link"
      href="https://doi.org/study pid"
      onBlur={[Function]}
      onFocus={[Function]}
      variant="inherit"
    >
      <MuiTypographyRoot
        as="a"
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        data-testid="isis-investigation-table-doi-link"
        href="https://doi.org/study pid"
        onBlur={[Function]}
        onFocus={[Function]}
        ownerState={
          Object {
            "align": "inherit",
            "children": "study pid",
            "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
            "color": "primary.main",
            "component": "a",
            "data-testid": "isis-investigation-table-doi-link",
            "gutterBottom": false,
            "href": "https://doi.org/study pid",
            "noWrap": false,
            "onBlur": [Function],
            "onFocus": [Function],
            "paragraph": false,
            "sx": Object {
              "color": "primary.main",
            },
            "variant": "inherit",
            "variantMapping": Object {
              "body1": "p",
              "body2": "p",
              "h1": "h1",
              "h2": "h2",
              "h3": "h3",
              "h4": "h4",
              "h5": "h5",
              "h6": "h6",
              "inherit": "p",
              "subtitle1": "h6",
              "subtitle2": "h6",
            },
          }
        }
        sx={
          Object {
            "color": "primary.main",
          }
        }
      >
        <Noop />
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
          data-testid="isis-investigation-table-doi-link"
          href="https://doi.org/study pid"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          study pid
        </a>
      </MuiTypographyRoot>
    </ForwardRef(Typography)>
  </MuiLinkRoot>
</ForwardRef(Link)>
`;

exports[`ISIS Investigations table component renders title, name and DOI as links in StudyHierarchy 1`] = `
<ForwardRef(Link)
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  data-testid="isis-investigations-table-title"
  to="/browseStudyHierarchy/instrument/4/study/5/investigation/1"
>
  <MuiLinkRoot
    className="MuiLink-root MuiLink-underlineAlways"
    color="primary"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    data-testid="isis-investigations-table-title"
    onBlur={[Function]}
    onFocus={[Function]}
    ownerState={
      Object {
        "children": "Test 1",
        "color": "primary",
        "component": Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        },
        "data-testid": "isis-investigations-table-title",
        "focusVisible": false,
        "to": "/browseStudyHierarchy/instrument/4/study/5/investigation/1",
        "underline": "always",
        "variant": "inherit",
      }
    }
    to="/browseStudyHierarchy/instrument/4/study/5/investigation/1"
    variant="inherit"
  >
    <Noop />
    <ForwardRef(Typography)
      className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-testid="isis-investigations-table-title"
      onBlur={[Function]}
      onFocus={[Function]}
      to="/browseStudyHierarchy/instrument/4/study/5/investigation/1"
      variant="inherit"
    >
      <MuiTypographyRoot
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        data-testid="isis-investigations-table-title"
        onBlur={[Function]}
        onFocus={[Function]}
        ownerState={
          Object {
            "align": "inherit",
            "children": "Test 1",
            "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
            "color": "primary.main",
            "component": Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
            "data-testid": "isis-investigations-table-title",
            "gutterBottom": false,
            "noWrap": false,
            "onBlur": [Function],
            "onFocus": [Function],
            "paragraph": false,
            "sx": Object {
              "color": "primary.main",
            },
            "to": "/browseStudyHierarchy/instrument/4/study/5/investigation/1",
            "variant": "inherit",
            "variantMapping": Object {
              "body1": "p",
              "body2": "p",
              "h1": "h1",
              "h2": "h2",
              "h3": "h3",
              "h4": "h4",
              "h5": "h5",
              "h6": "h6",
              "inherit": "p",
              "subtitle1": "h6",
              "subtitle2": "h6",
            },
          }
        }
        sx={
          Object {
            "color": "primary.main",
          }
        }
        to="/browseStudyHierarchy/instrument/4/study/5/investigation/1"
      >
        <Noop />
        <Link
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
          data-testid="isis-investigations-table-title"
          onBlur={[Function]}
          onFocus={[Function]}
          to="/browseStudyHierarchy/instrument/4/study/5/investigation/1"
        >
          <LinkAnchor
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            data-testid="isis-investigations-table-title"
            href="/browseStudyHierarchy/instrument/4/study/5/investigation/1"
            navigate={[Function]}
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              data-testid="isis-investigations-table-title"
              href="/browseStudyHierarchy/instrument/4/study/5/investigation/1"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Test 1
            </a>
          </LinkAnchor>
        </Link>
      </MuiTypographyRoot>
    </ForwardRef(Typography)>
  </MuiLinkRoot>
</ForwardRef(Link)>
`;

exports[`ISIS Investigations table component renders title, name and DOI as links in StudyHierarchy 2`] = `"Test 1"`;

exports[`ISIS Investigations table component renders title, name and DOI as links in StudyHierarchy 3`] = `
<ForwardRef(Link)
  data-testid="isis-investigation-table-doi-link"
  href="https://doi.org/study pid"
>
  <MuiLinkRoot
    className="MuiLink-root MuiLink-underlineAlways"
    color="primary"
    component="a"
    data-testid="isis-investigation-table-doi-link"
    href="https://doi.org/study pid"
    onBlur={[Function]}
    onFocus={[Function]}
    ownerState={
      Object {
        "children": "study pid",
        "color": "primary",
        "component": "a",
        "data-testid": "isis-investigation-table-doi-link",
        "focusVisible": false,
        "href": "https://doi.org/study pid",
        "underline": "always",
        "variant": "inherit",
      }
    }
    variant="inherit"
  >
    <Noop />
    <ForwardRef(Typography)
      className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
      color="primary"
      component="a"
      data-testid="isis-investigation-table-doi-link"
      href="https://doi.org/study pid"
      onBlur={[Function]}
      onFocus={[Function]}
      variant="inherit"
    >
      <MuiTypographyRoot
        as="a"
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        data-testid="isis-investigation-table-doi-link"
        href="https://doi.org/study pid"
        onBlur={[Function]}
        onFocus={[Function]}
        ownerState={
          Object {
            "align": "inherit",
            "children": "study pid",
            "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
            "color": "primary.main",
            "component": "a",
            "data-testid": "isis-investigation-table-doi-link",
            "gutterBottom": false,
            "href": "https://doi.org/study pid",
            "noWrap": false,
            "onBlur": [Function],
            "onFocus": [Function],
            "paragraph": false,
            "sx": Object {
              "color": "primary.main",
            },
            "variant": "inherit",
            "variantMapping": Object {
              "body1": "p",
              "body2": "p",
              "h1": "h1",
              "h2": "h2",
              "h3": "h3",
              "h4": "h4",
              "h5": "h5",
              "h6": "h6",
              "inherit": "p",
              "subtitle1": "h6",
              "subtitle2": "h6",
            },
          }
        }
        sx={
          Object {
            "color": "primary.main",
          }
        }
      >
        <Noop />
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
          data-testid="isis-investigation-table-doi-link"
          href="https://doi.org/study pid"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          study pid
        </a>
      </MuiTypographyRoot>
    </ForwardRef(Typography)>
  </MuiLinkRoot>
</ForwardRef(Link)>
`;
